src/osfiber_aarch64.c

```diff
@@ -23,12 +25,23 @@ void marl_fiber_trampoline(void (*target)(void*), void* arg) {
   target(arg);
 }
 
+// This is needed for HWSAan runtimes that don't have this commit:
+// https://reviews.llvm.org/D149228.
+__attribute__((weak)) void __hwasan_tag_memory(const volatile void *p,
+    unsigned char tag, size_t size);
+__attribute((weak)) void *__hwasan_tag_pointer(const volatile void *p,
+    unsigned char tag);
+
 MARL_EXPORT
 void marl_fiber_set_target(struct marl_fiber_context* ctx,
                            void* stack,
                            uint32_t stack_size,
                            void (*target)(void*),
                            void* arg) {
+  if (__hwasan_tag_memory && __hwasan_tag_pointer) {
```

Comment:
Is there a preprocessor define we can use to guard this?

Reply:
Not really, on Android it is possible to run an instrumented libc (which returns tagged pointers) even if your program is compiled as normal, so we won't be able to know whether we are running with HWASan at compile time.

Reply:
Fair enough.